addons/account_peppol/tests/test_peppol_messages.py

```diff
@@ -127,29 +125,39 @@ def _patch_peppol_requests(self, error=False):
                 ],
             }}
         }
+        return proxy_documents, responses
 
-        def _mocked_post(url, *args, **kwargs):
-            response = Mock()
-            response.status_code = 200
-            url = url.split('api')[1]
-            if url.endswith('api/peppol/1/send_document'):
-                if not kwargs['json']['params']['documents']:
-                    raise UserError('No documents were provided')
-
-            if url == '/peppol/1/get_document':
-                uuid = kwargs['json']['params']['message_uuids'][0]
-                response.json = lambda: {
-                    'result': {uuid: proxy_documents[uuid]}}
-                return response
+    @contextmanager
+    def _set_context(self, other_context):
+        previous_context = self.env.context
+        self.env.context = dict(previous_context, **other_context)
+        yield self
+        self.env.context = previous_context
```

Comment:
That part looks ugly. Is it not enough to just add `with_context` manually in the 3 tests needing an extra key inside the context?

Reply:
(necessary because _request_handler is getting only the global context)